core/templates/dev/head/pages/collection_editor/settings_tab/CollectionPermissionsCardDirective.js

```diff
@@ -22,10 +22,25 @@ oppia.directive('collectionPermissionsCard', [function() {
     restrict: 'E',
     templateUrl: 'inline/collection_permissions_card_directive',
     controller: [
-      '$scope',
-      function($scope) {
-        $scope.ownerNames = GLOBALS.ownerNames;
-        $scope.isPrivate = GLOBALS.isPrivate;
+      '$scope', 'CollectionRightsBackendApiService',
+      'EVENT_COLLECTION_STATUS_CHANGE',
+      function($scope, CollectionRightsBackendApiService,
+        EVENT_COLLECTION_STATUS_CHANGE) {
+        $scope.rightsHasLoaded = false;
+
+        var refreshPermissionsCard = function() {
+          $scope.rightsHasLoaded = false;
```

Comment:
I think it would be weird for the permissions card to disappear in the middle, so I wouldn't reset this every time an update happens to the permissions.